### PR TITLE
questions and answer questions page

### PR DESCRIPTION
**Pull request by xuwhite**
- success color join angeltype button
- differentiate Answer questions from questions page
- add other contact options to questions page
- Password reset in User state

### CODE REVIEW

File: config/config.default.php [hunk 1]
```diff
@@ -53,6 +53,15 @@
         'Contact' => env('CONTACT_EMAIL', 'mailto:ticket@c3heaven.de'),
     ],
 
+    // Other contact points then ask the heaven
+    // You can set as many contact points as you like analog to email some
```
Comment (MyIgel) [inline config/config.default.php:57]: I am missing some more structure in that description

File: config/config.default.php [hunk 2]
```diff
@@ -53,6 +53,15 @@
...
+    // suggestions are dect number or mastodon account
+    // To disable a contact option in the config.php, you can set its value to null, if all are null
```
Comment (MyIgel) [inline config/config.default.php:59]: then what?

File: config/config.default.php [hunk 3]
```diff
@@ -53,6 +53,15 @@
...
+    'contact_options' => [
+        // email address
+        'E-mail' => env('CONTACT_EMAIL', 'mailto:ticket@c3heaven.de'),
```
Comment (MyIgel) [inline config/config.default.php:62]: `general.email` might be a better option here?

File: resources/lang/de_DE/default.po [hunk 4]
```diff
@@ -1872,6 +1875,9 @@ msgstr "Antwort"
 msgid "question.delete.title"
 msgstr "Frage \"%s\" löschen"
 
+msgid "questions.contact_options"
+msgstr "Andere Kontaktmöglichkeiten: "
```
Comment (MyIgel) [inline resources/lang/de_DE/default.po:1879]: I would suggest `Weitere Kontaktmöglichkeiten` here

File: resources/views/pages/questions/overview.twig [hunk 5]
```diff
@@ -21,8 +51,17 @@
...
                             {% if has_permission_to('faq.view') %}
-                                <p>{{ __('question.faq_link', [url('/faq')]) | raw }}</p>
+                                {% if not is_admin | default(false) %}
+                                    {{ __('question.faq_link', [url('/faq')]) | raw }}
+                                {% endif %}
+                            {% endif %}
+                        {% endblock %}
+                        {%- block info_text %}
+                            <p>
+                            {% if config('contact_options') %}
```
Comment (MyIgel) [inline resources/views/pages/questions/overview.twig:61]: Should be hidden on the "answer questions" page

File: resources/views/pages/questions/overview.twig [hunk 6]
```diff
@@ -21,8 +51,17 @@
...
                         {% block faq_text %}
                             {% if has_permission_to('faq.view') %}
-                                <p>{{ __('question.faq_link', [url('/faq')]) | raw }}</p>
+                                {% if not is_admin | default(false) %}
+                                    {{ __('question.faq_link', [url('/faq')]) | raw }}
+                                {% endif %}
+                            {% endif %}
+                        {% endblock %}
+                        {%- block info_text %}
+                            <p>
```
Comment (MyIgel) [inline resources/views/pages/questions/overview.twig:60]: Should be inside the if block

File: config/config.default.php [hunk 7]
```diff
@@ -53,6 +53,15 @@
         'Contact' => env('CONTACT_EMAIL', 'mailto:ticket@c3heaven.de'),
     ],
 
+    // Other contact points then ask the heaven
+    // You can set as many contact points as you like analog to email some
+    // suggestions are dect number or mastodon account
+    // To disable a contact option in the config.php, you can set its value to null, if all are null
+    'contact_options' => [
+        // email address
```
Comment (MyIgel) [inline config/config.default.php:61]: Shouldn't that be at least `E-mail`? ;)

File: resources/lang/de_DE/default.po [hunk 8]
```diff
@@ -1872,6 +1875,9 @@ msgstr "Antwort"
 msgid "question.delete.title"
 msgstr "Frage \"%s\" löschen"
 
+msgid "questions.contact_options"
```
Comment (MyIgel) [inline resources/lang/de_DE/default.po:1878]:
```suggestion
msgid "question.contact_options"
```
to fit the other keys

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: config/config.default.php (hunk 1 above)
Comment (xuwhite) [reply config/config.default.php:57]: do you have any specific proposals?

File: resources/views/pages/questions/overview.twig (hunk 6 above)
Comment (xuwhite) [reply resources/views/pages/questions/overview.twig:60]: nope because it looks funny if its not shown

Comment (MyIgel) [reply resources/views/pages/questions/overview.twig:60]: then that should be fixed with css and not by adding empty paragraphs

File: config/config.default.php (hunk 1 above)
Comment (MyIgel) [reply config/config.default.php:57]: Maybe something like
```
    // Other ways to ask the heaven
    // Multiple contact options / links are possible, analogue to footer_items
```

File: resources/views/pages/questions/overview.twig (hunk 6 above)
Comment (xuwhite) [reply resources/views/pages/questions/overview.twig:60]: 😝

File: resources/views/pages/questions/overview.twig (hunk 5 above)
Comment (MyIgel) [reply resources/views/pages/questions/overview.twig:62]: @xuwhite this is still an issue?

Comment (xuwhite) [reply resources/views/pages/questions/overview.twig:62]: nopw its not

File: includes/view/User_view.php [hunk 9]
```diff
@@ -765,12 +766,15 @@ function User_view_state_user($user_source)
  * @param User $user_source
  * @return array
  */
-function User_view_state_admin($freeloader, $user_source)
+function User_view_state_admin($freeloader, $user_source,)
```
Comment (MyIgel) [inline includes/view/User_view.php:769]:
```suggestion
function User_view_state_admin($freeloader, $user_source)
```

File: resources/views/pages/questions/overview.twig [hunk 10]
```diff
@@ -21,7 +51,18 @@
...
-                                <p>{{ __('question.faq_link', [url('/faq')]) | raw }}</p>
+                                {% if not is_admin | default(false) %}
+                                    <p>
+                                        {{ __('question.faq_link', [url('/faq')]) | raw }}
+                                    </p>
+                                {% endif %}
+                            {% endif %}
+                        {% endblock %}
+                        {%- block info_text %}
+                            {% if config('contact_options') %}
```
Comment (MyIgel) [inline resources/views/pages/questions/overview.twig:62]: Screenshot_20240225_114155

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: resources/views/pages/questions/overview.twig [hunk 11]
```diff
@@ -21,7 +51,18 @@
...
-                                <p>{{ __('question.faq_link', [url('/faq')]) | raw }}</p>
+                                {% if not is_admin | default(false) %}
+                                    <p>
+                                        {{ __('question.faq_link', [url('/faq')]) | raw }}
+                                    </p>
+                                {% endif %}
+                            {% endif %}
+                        {% endblock %}
+                        {%- block info_text %}
+                            {% if config('contact_options') and not is_admin %}
```
Comment (MyIgel) [inline resources/views/pages/questions/overview.twig:62]:
```
 Twig \ Error \ RuntimeError
Variable "is_admin" does not exist.
```

**Review by MyIgel — CHANGES_REQUESTED** [review]

**Pull request merged**